src/main/java/com/google/sps/servlets/NewMessageServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.servlets;
+import com.google.sps.data.Chat;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+// import com.google.cloud.datastore.Query;
+// import com.google.cloud.datastore.QueryResults;
+// import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.IncompleteKey;
+import com.google.cloud.datastore.Value;
+import com.google.cloud.datastore.ListValue;
+import com.google.cloud.datastore.StringValue;
+import com.google.gson.Gson;
+import com.google.sps.data.Chat;
+
+import java.io.IOException;
+// import java.util.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+// import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+// import org.jsoup.Jsoup;
+// import org.jsoup.safety.Whitelist;
+import java.lang.String;
+import com.google.sps.data.Chat;
+
+// Try typing "/chats" or "/chats?userid=Elijah" or "/chats?chatid=someid"
+// to see how the query parameters work for the servlet
+@WebServlet("/newmessage")
```

Comment:
FYI it's a little too specific to have an endpoint like for a single CRUD operation. `POST /messages` already indicates the creation of a new message. Using `/messages` also lets you add reasonable handlers for getting/updating/deleting messages too later.

Reply:
Ok Ill change that